Sources/TuistKit/Graph/GraphLoader.swift

```diff
@@ -42,13 +42,26 @@ class GraphLoader: GraphLoading {
         return graph
     }
 
-    func loadWorkspace(path: AbsolutePath) throws -> Graph {
+    func loadWorkspace(path: AbsolutePath) throws -> (Workspace, Graph) {
         let cache = GraphLoaderCache()
         let circularDetector = GraphCircularDetector()
         let workspace = try modelLoader.loadWorkspace(at: path)
-        let projects = try workspace.projects.map { (projectPath) -> (AbsolutePath, Project) in
-            try (projectPath, Project.at(projectPath, cache: cache, circularDetector: circularDetector, modelLoader: modelLoader))
+
+        func traverseProjects(element: Workspace.Element) throws -> [(AbsolutePath, Project)] {
```

Comment:
Nitpick: What about extracting this function into the class. Having nested functions makes the code less readable.